modules/ts/src/ts_perf.cpp

```diff
@@ -1359,8 +1359,10 @@ bool TestBase::next()
         lastActivityPrintTime = 0;
         metrics.clear();
     }
-
+    
+#ifdef OPENCV_HAVE_THREAD_SUPPORT
     cv::theRNG().state = param_seed; //this rng should generate same numbers for each run
+#endif
```

Comment:
We still need to reset RNG state to initial value.
There is nothing about the threading.

All TLS containers must work, with behavior of regular variables.